nodes/persistence-node/src/services/CacheRunner.ts

```diff
@@ -68,83 +75,71 @@ export class CacheRunner {
     this.deps.storage.lastBlockNumber = resp.toBlock;
     await this.deps.storage.save();
   }
-
-  async processUnresponsive() {
-    this.deps.logger.log("Processing unresponsive packages...");
-    
-    const ensNodes = Object.keys(this.deps.storage.unresponsiveEnsNodes);
-    this.deps.storage.unresponsiveEnsNodes = {};
-
-    await this.processEnsNodes(ensNodes);
-  }
-
-  async processEnsIpfs(ensNode: string, ipfsHash: string | undefined): Promise<boolean> {
+  
+  async processEnsIpfs(ensNode: string, ipfsHash: string | undefined): Promise<ProcessEnsIpfsStatus> {
     const ensIpfsCache = this.deps.storage.ensIpfs;
     const ipfsEnsCache = this.deps.storage.ipfsEns;
 
-    if(!ipfsHash) {
+    if (!ipfsHash) {
       const savedIpfsHash = ensIpfsCache[ensNode];
 
-      if(savedIpfsHash) {
+      if (savedIpfsHash) {
         this.deps.logger.log("ENS no longer points to an IPFS hash");
         this.deps.logger.log("Unpinning...");
 
         const success = await unpinCid(this.deps.ipfsNode, this.deps.ipfsConfig, savedIpfsHash);
         
-        if(success) {
-
-          if(ipfsEnsCache[savedIpfsHash]) {
+        if (success) {
+          if (ipfsEnsCache[savedIpfsHash]) {
             delete ipfsEnsCache[savedIpfsHash];
           }
 
           delete ensIpfsCache[ensNode];
 
           this.deps.logger.log("Unpinned successfully");
+          return ProcessEnsIpfsStatus.Unpinned;
         } else {
           this.deps.logger.log("Unpinning failed");
+          this.deps.storage.unresponsiveEnsNodes[ensNode] = true;
```

Comment:
I am not sure if this should be treated as unresponsive node, so I decided to add it and open a discussion on pull request.